static/js/util/validation.js

```diff
@@ -135,7 +142,8 @@ export function employmentValidation(profile) {
     let requiredFields = handleNestedValidation(profile, nestedKeys, 'work_history');
     let errors = checkFieldPresence(profile, requiredFields, messages);
 
-    for (let [index, workHistory] of Object.entries(profile.work_history)) {
+    for (let entry: WorkEntry of Object.entries(profile.work_history)) {
+      let index = entry[0], workHistory = entry[1];
```

Comment:
It looks like you can simplify this to `let [index, workHistory] = entry` at least